### ISSUE TITLE
Semi-automated masking for depth mapping.

### ISSUE THREAD

**Issue by ghost** [cdcseacave/openMVS#791]
Manually masking out sections not to be processed for hundreds of images is very time consuming.

Proposed outline:
1. Input obb or bring up the viewer to select a region with gizmos.
2. Create a rough mesh using a low resolution dense point cloud or sparse point cloud.
   * Trimming the point cloud before meshing.
4. Generate and export masks to the images folder with an option to expand the mask.

Additional notes:
* Hidden option under ReconstructMesh?
   -? [ --??? ] arg (=0) obb input for mask generation (0 - disabled, 1 - use obb file, 2 - open viewer, ~~3 - auto~~)
   * Opening the viewer shows the sparse point cloud.
* Users could use the undistorted masks and images starting from SfM.

![semi_automated_masking](https://user-images.githubusercontent.com/32020441/159367189-ca8abc87-637d-4e9a-a1e1-1c46144aa131.png)
![Geoxor_Virtual_Arcadia_2020_MIX](https://user-images.githubusercontent.com/32020441/159411314-7e83ee28-5e1c-4e93-8073-06ef4b603f01.png)

**Comment by SSground** [comment]
Can you elaborate on which commands to use to achieve，thanks

**Comment by cdcseacave** [comment]
Yes, using the sparse point cloud as an initial rough mesh to generate a mask in each image is a good idea. I can do it when I'll find some time, I am busy with my job!

**Comment by 3XH6R** [comment]
Test run for masking using what's in commit 168bbdc
Dataset: [RealityCapture Small Cottage](https://www.capturingreality.com/free-datasets)
https://www.youtube.com/watch?v=Swgyh-j-ryg

![uwu](https://user-images.githubusercontent.com/32020441/160269104-518fe475-29a8-4bbd-850d-2643f244a47d.png)

**Comment by cdcseacave** [comment]
Semi-automated masking is outside current scope. Manual masks can be used by placing \_mask.png\ files alongside images.
